src/main/java/uk/nhs/digital/nhsconnect/lab/results/inbound/NhsAckProducerService.java

```diff
@@ -0,0 +1,159 @@
+package uk.nhs.digital.nhsconnect.lab.results.inbound;
+
+import com.github.mustachejava.Mustache;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.nhs.digital.nhsconnect.lab.results.mesh.message.WorkflowId;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Interchange;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.NhsAckContent;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.NhsAckStatus;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.NhsAckMessageContent;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.InterchangeHeader;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.MessageHeader;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.InterchangeParsingException;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.MessageParsingException;
+import uk.nhs.digital.nhsconnect.lab.results.sequence.SequenceService;
+import uk.nhs.digital.nhsconnect.lab.results.utils.TimestampService;
+import uk.nhs.digital.nhsconnect.lab.results.inbound.MessageProcessingResult.Success;
+import uk.nhs.digital.nhsconnect.lab.results.inbound.MessageProcessingResult.Error;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
+import static uk.nhs.digital.nhsconnect.lab.results.utils.TemplateUtils.loadTemplate;
+import static uk.nhs.digital.nhsconnect.lab.results.utils.TemplateUtils.fillTemplate;
+
+@Component
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
+public class NhsAckProducerService {
+
+    @Autowired
```

Comment:
I don't think `@Autowired` is needed on these fields, as it's applied to the constructor